forc-plugins/forc-client/src/util/account.rs

```diff
@@ -24,33 +24,14 @@ pub enum ForcClientAccount {
 
 #[async_trait]
 impl Account for ForcClientAccount {
-    async fn get_asset_inputs_for_amount(
-        &self,
-        asset_id: AssetId,
-        amount: u64,
-        excluded_coins: Option<Vec<CoinTypeId>>,
-    ) -> Result<Vec<Input>> {
-        match self {
-            ForcClientAccount::Wallet(wallet) => {
-                wallet
-                    .get_asset_inputs_for_amount(asset_id, amount, excluded_coins)
-                    .await
-            }
-            ForcClientAccount::KmsSigner(account) => {
-                account
-                    .get_asset_inputs_for_amount(asset_id, amount, excluded_coins)
-                    .await
-            }
-        }
-    }
-
     fn add_witnesses<Tb: TransactionBuilder>(&self, tb: &mut Tb) -> Result<()> {
         tb.add_signer(self.clone())?;
 
         Ok(())
     }
 }
 
+#[async_trait]
```

Comment:
why did we need to introduce this?

Reply:
required since traits cannot have async functions inside them - or we need to explicitly define the return type using the future syntax.

Looking at the trait definition though:
```rust
#[cfg_attr(not(target_arch = "wasm32"), async_trait)]
pub trait ViewOnlyAccount: std::fmt::Debug + Send + Sync + Clone {
```
I think i should add the same same proc macro.

Reply:
actually ill leave as is; since the account impl above could also use the same change; ill keep them in sync